### PR TITLE
Adds breadcrumb component

### PR DESCRIPTION
**Pull request by samiwel**
## Description
This PR adds a breadcrumb component as per the Sketch designs.
**N.B - This pull request is based off the `sidebar-component` branch. Please ensure that the `sidebar-component` branch is merged Before this this branch.**

## How to test
Take a look at both `Breadcrumb` and `BreadcrumbLink` components in React StoryBook.

## Screenshot
image

### CODE REVIEW

File: src/components/BreadcrumbLink/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  text-decoration: none;
+  color: ${colorBlue};
+
+  &:hover {
+    color: ${colorLightBlue}
+  }
+`
+
+const ChevronIcon = styled(Chevron)`
+  margin: 0 8px 0 8px;
```
Comment (hamishtaplin) [inline src/components/BreadcrumbLink/index.js]: don't use `px` units, prefer `ems`.

File: src/components/BreadcrumbLink/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import {colorBlue, colorLightBlue} from 'constants/theme'
+import tinycolor from 'tinycolor2'
```
Comment (hamishtaplin) [inline src/components/BreadcrumbLink/index.js]: I don't think this gets used?

File: src/components/BreadcrumbLink/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const BreadcrumbLink = styled.a`
+  font-size: 11px;
+
+  ${props => props.href && Clickable}
+`
+
+export default (props) =>
+  <BreadcrumbLink {...props}>
+    {props.children}
+    {props.parent &&
```
Comment (hamishtaplin) [inline src/components/BreadcrumbLink/index.js]: Can the logic for the separator be handled just by looping over the children instead? Would negate having to use a prop for this.

File: src/components/BreadcrumbLink/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    color: ${colorLightBlue}
+  }
+`
+
+const ChevronIcon = styled(Chevron)`
+  margin: 0 8px 0 8px;
+`
+
+const BreadcrumbLink = styled.a`
+  font-size: 11px;
```
Comment (hamishtaplin) [inline src/components/BreadcrumbLink/index.js]: Use `ems` (don't worry about it being exactly 11px)

File: src/components/BreadcrumbLink/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import tinycolor from 'tinycolor2'
+import Chevron from './chevron'
+
+const Clickable = css`
+  cursor: pointer;
+  text-decoration: none;
+  color: ${colorBlue};
+
+  &:hover {
+    color: ${colorLightBlue}
```
Comment (hamishtaplin) [inline src/components/BreadcrumbLink/index.js]: Add underline on hover

**Comment by hamishtaplin** [comment]
For the sake of consistency, can you bundle the two (related) components up into a folder like I have done with the Tabs and Sidebar components? It's a pain but it gives a cleaner interface for `import` as you can do it one line.

**Comment by hamishtaplin** [comment]
screenshot 2017-03-30 15 58 07

the svg needs tidying up a bit, attributes should be camel-cased

File: src/components/Breadcrumb/story.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import { storiesOf } from '@kadira/storybook'
+import Breadcrumb from 'components/Breadcrumb'
+
+// This will be replaced by React-Router links
+const Link = (props) => (
+  <a>{props.text}</a>
```
Comment (hamishtaplin) [inline src/components/Breadcrumb/story.js]: needs a `href='#` or something or hover styles don't work

File: src/components/Breadcrumb/index.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import {breadcrumbBackground} from 'constants/theme'
+import tinycolor from 'tinycolor2'
+import Chevron from './chevron'
+
+const ChevronIcon = styled(Chevron)`
+  margin: 0 8px 0 8px;
```
Comment (hamishtaplin) [inline src/components/Breadcrumb/index.js]: Don't use `px` for anything (always ems/rems). These margins should be a bit smaller too.

File: src/components/Breadcrumb/index.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+`
+
+const Breadcrumb = styled.div`
+  background-color: ${breadcrumbBackground};
+  min-height: 28px;
```
Comment (hamishtaplin) [inline src/components/Breadcrumb/index.js]: This shouldn't be necessary if you use line-height to control the height of the text within. Then use top/bottom padding on this container to make sure.

Make sure everything is aligned vertically and spacing is consistent above/below the text.

File: src/components/Breadcrumb/index.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import {breadcrumbBackground} from 'constants/theme'
```
Comment (hamishtaplin) [inline src/components/Breadcrumb/index.js]: This colour is used elsewhere and so should be `colorGrey` or something

**Comment by samiwel** [comment]
@hamishtaplin Fixed the link styles for now. The links will be replaced with React Router Links once both PRs are merged.

**Pull request merged**